src/index.ts

```diff
@@ -3,64 +3,98 @@ import path from 'path'
 import { homedir } from 'os'
 import deepmerge from 'deepmerge'
 import { extendConfig } from 'hardhat/config'
-import { HardhatConfig, NetworkConfig, NetworksConfig, HardhatUserConfig } from 'hardhat/types'
+import { HardhatConfig, NetworkConfig, NetworksUserConfig, NetworkUserConfig, HardhatUserConfig } from 'hardhat/types'
+import { HardhatPluginError } from 'hardhat/plugins'
+import { parseLocalNetworksConfigPath } from './utils'
 
 const HARDHAT_CONFIG_DIR = '.hardhat'
 const HARDHAT_NETWORK_CONFIG_FILE = 'networks.json'
 
 export interface LocalNetworksConfig {
-  networks: NetworksConfig
-  defaultConfig: NetworkConfig
+  networks?: NetworksUserConfig
+  defaultConfig?: NetworkUserConfig
+}
+
+interface LocalNetworksConfigInternal extends LocalNetworksConfig {
+  networks: NetworksUserConfig
+  defaultConfig: NetworkUserConfig
 }
 
 extendConfig((hardhatConfig: HardhatConfig, userConfig: HardhatUserConfig): void => {
-  const localNetworksConfig = readLocalNetworksConfig(userConfig)
+  const homeLocalNetworksConfig = readHomeNetworksConfig()
+  const localNetworksConfig = readLocalNetworksConfig(hardhatConfig, userConfig)
 
   const userNetworkConfigs = userConfig.networks || []
   Object.entries(userNetworkConfigs).forEach(([networkName, userNetworkConfig]) => {
     hardhatConfig.networks[networkName] = (deepmerge.all([
       hardhatConfig.networks[networkName] || {},
+      homeLocalNetworksConfig.defaultConfig,
+      homeLocalNetworksConfig.networks[networkName] || {},
       localNetworksConfig.defaultConfig,
+      userNetworkConfig as object,
       localNetworksConfig.networks[networkName] || {},
```

Comment:
This is a bit weird, cause it seems you're prioritizing specific network configs of the local config file over the project's config, but not the default config in the local config file over the project's config.